src/drivers/fixed-clock.c

```diff
@@ -7,16 +7,22 @@
 
 #include <metal/drivers/fixed-clock.h>
 #include <stddef.h>
+#include <metal/machine.h>
 
 long __metal_driver_fixed_clock_get_rate_hz(const struct metal_clock *gclk)
 {
     const struct __metal_driver_fixed_clock *clk = (void *)gclk;
-    return clk->rate;
+    return __metal_driver_fixed_clock_rate();
```

Comment:
This is missing the clock pointer.

Reply:
You right, should add one since custom and/or future design way have many clocks to select from